release/models/virtual-devices/openconfig-virtual-devices.yang

```diff
@@ -0,0 +1,386 @@
+      description
+        "The id/name of the virtual device";
+    }
+
+    leaf type {
+      type identityref {
+        base oc-vdevice-types:VIRTUAL_DEVICE_TYPE;
+      }
+      mandatory true;
+      description
+        "The type of virtual device";
+    }
+
+    leaf description {
+      type string;
+      description
+        "A textual description of the virtual device.";
+    }
+
+    leaf enabled {
+      type boolean;
+      default "true";
+      description
+        "This leaf contains the configured, desired state of the
+        virtual device.";
+    }
+
+    leaf instance {
+      type string;
+      description
+        "Instance to associate with this device";
+    }
+
+    leaf device-version {
+      type string;
+      description
+        "Device specific version
+        
+        For e.g.: 0.95 or 1.0 for virtio";
+    }
+
+    uses vdevice-model-version;
+  }
+
+  grouping vdevice-common-state {
+    description
+      "Operational state data (in addition to intended configuration)
+      at the global level for this virtual device";
+
+    oc-ext:operational;
+
+    leaf admin-status {
+      type enumeration {
+        enum UP {
+          description
+            "Ready for the compute instance to use.";
+        }
+        enum DOWN {
+          description
+            "Admin of the infrastructure has disabled this device.";
+        }
+        enum ERROR {
+          description
+            "Device error detected which has disabled this device.";
+        }
+      }
+      mandatory true;
+      description
+        "The desired state of the virtual device.
+        Reflects the administrative state as set by enabling or
+        disabling the interface.";
+      oc-ext:telemetry-on-change;
+    }
+
+    leaf oper-status {
+      type enumeration {
+        enum UP {
+          value 1;
+          description
+            "Ready for the compute instance to use.";
+        }
+        enum DOWN {
+          value 2;
+          description
+            "Admin of the infrastructure has disabled this device.";
+        }
+        enum TESTING {
+          value 3;
+          description
+            "Device error detected which has disabled this device.";
+        }
+        enum UNKNOWN {
+          value 4;
+          description
+            "Status cannot be determined for some reason.";
+        }
+        enum DORMANT {
+          value 5;
+          description
+            "Waiting for some external event.";
+        }
+        enum NOT_PRESENT {
+          value 6;
+          description
+            "Some component (typically hardware) is missing.";
+        }
+        enum LOWER_LAYER_DOWN {
+          value 7;
+          description
+            "Down due to state of lower-layer interface(s).";
+        }
+      }
+      mandatory true;
+      description
+        "The current operational state of the virtual device.";
+      oc-ext:telemetry-on-change;
+    }
+
+    leaf last-change {
```

Comment:
If not absolutely needed, can we leave this out for now?

Reply:
This is needed for alignment with interfaces